pkg/logging/log.go

```diff
@@ -43,13 +44,17 @@ func Init(flags *flag.FlagSet) {
 
 // Setup configures the logger with the supplied log format.
 // It returns an error if the JSON logger could not be initialized or passed logFormat is not recognized.
-func Setup(logFormat string) error {
+func Setup(logFormat string, level int) error {
 	switch logFormat {
 	case TextFormat:
 		// in text mode we use FormatSerialize format
 		globalLog = klogr.New()
 	case JSONFormat:
-		zapLog, err := zap.NewProduction()
+		zc := zap.NewProductionConfig()
+		// Zap's levels get more and less verbose as the number gets smaller and higher respectively (DebugLevel is -1, InfoLevel is 0, WarnLevel is 1, and so on).
+		zc.Level = zap.NewAtomicLevelAt(zapcore.Level(-1 * level))
```

Comment:
Why do we set a negative value ?

Reply:
this is because due to how zap is implemented. it multiplies the value passed with -1 internally. according to the docs

```
Zap uses semantically named levels for logging (DebugLevel, InfoLevel, WarningLevel, ...).
Logr uses arbitrary numeric levels. By default logr's V(0) is zap's InfoLevel and V(1) is 
zap's DebugLevel (which is numerically -1). Zap does not have named levels that are 
more verbose than DebugLevel, but it's possible to fake it.
```
```go
zc := zap.NewProductionConfig()
    zc.Level = zap.NewAtomicLevelAt(zapcore.Level(-2))
    z, err := zc.Build()
    if err != nil {
```
```
Zap's levels get more verbose as the number gets smaller and more important
and the number gets larger (DebugLevel is -1, InfoLevel is 0, WarnLevel is 1, and so on).
The -2 in the above snippet means that log.V(2).Info() calls will be active. -3 would enable log.V(3).Info()
```

Reply:
Not sure I completely understand but we can always fix it later.